src/PowerShellEditorServices.Host/CodeLens/PesterCodeLensProvider.cs

```diff
@@ -63,12 +72,40 @@ public PesterCodeLensProvider(EditorSession editorSession)
                     new ClientCommand(
                         "PowerShell.RunPesterTests",
                         "Debug tests",
-                        new object[] { scriptFile.ClientFilePath, true /* Run in debugger */, pesterSymbol.TestName })),
+                        new object[] { scriptFile.ClientFilePath, true /* Run in debugger */, pesterSymbol.TestName, describeBlockLineNumber })),
             };
 
             return codeLensResults;
         }
 
+        /// <summary>
+        /// Used to determine the value of <see cref="_pesterV4_6_0_OrHigherAvailable"/> as a background task.
+        /// </summary>
+        private static bool DeterminePesterVersion()
+        {
+            using (var powerShell = System.Management.Automation.PowerShell.Create())
```

Comment:
Here you should use the `PSCommand` class I think. This is our abstraction over the PowerShell object to reuse runspaces.

Correct me if I'm wrong @rkeithhill & @SeeminglyScience

Reply:
In addition, PowerShellContext has quite a few `Execute` methods. Most are async but there should be some sync versions that execute commands.

Reply:
Ok, I was thinking of passing a `PowerShellContext` to the constructor as I have seen that other places use that for the runspace. I did not know about `PSCommand`, sounds like an additional improvement (that I might look into applying for PSSA as well). Even if there are only async methods, we can still just call `Task.Wait()` to make it sync